### PR TITLE
Following/Followers on profile

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Loads trackers/tracking (tracking is from 'you' for now, working with CORE to get that working)
- [x] You can click on people and push new profiles onto the stack, you get  back button also in that state

Stuff for later
- [ ] Follow/following state on avatar

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 9252cf8cd47d...ea9f14b4c39d introduced some visual changes:
 * Changed: **Profile-Changed** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466623074696-9252cf8cd47d-ea9f14b4c39d/Profile-Changed-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466623074696-9252cf8cd47d-ea9f14b4c39d/Profile-Changed-ea9f14b4c39d.png)  
   Profile-Changed comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 9252cf8cd47d...3f82b7071983 introduced some visual changes:
 * Changed: **Profile-Unfollowed** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Profile-Unfollowed-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Profile-Unfollowed-3f82b7071983.png)  
   Profile-Unfollowed comparison
 * Changed: **Profile-Followed** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Profile-Followed-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Profile-Followed-3f82b7071983.png)  
   Profile-Followed comparison
 * Changed: **Profile-Changed** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Profile-Changed-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Profile-Changed-3f82b7071983.png)  
   Profile-Changed comparison
 * Changed: **Menubar-Public_Truncated** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Menubar-Public_Truncated-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Menubar-Public_Truncated-3f82b7071983.png)  
   Menubar-Public_Truncated comparison
 * Changed: **Menubar-Private_Truncated** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Menubar-Private_Truncated-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Menubar-Private_Truncated-3f82b7071983.png)  
   Menubar-Private_Truncated comparison
 * Changed: **Menubar-Private_Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Menubar-Private_Normal-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Menubar-Private_Normal-3f82b7071983.png)  
   Menubar-Private_Normal comparison
 * Changed: **Menubar-Private_Menu_Showing** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Menubar-Private_Menu_Showing-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466628208930-9252cf8cd47d-3f82b7071983/Menubar-Private_Menu_Showing-3f82b7071983.png)  
   Menubar-Private_Menu_Showing comparison

File: shared/actions/tracker.js [hunk 1]
```diff
@@ -73,29 +74,15 @@ export function registerTrackerChangeListener (): TrackerActionCreator {
   }
 }
 
-export function registerUserChangeListener (): TrackerActionCreator {
-  return dispatch => {
-    const params: incomingCallMapType = {
-      'keybase.1.NotifyUsers.userChanged': ({uid}) => {
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: this causes spurious popups, removing for now

File: shared/actions/tracker.js [hunk 2]
```diff
@@ -104,13 +91,35 @@ export function getMyProfile (): TrackerActionCreator {
...
+      dispatch(triggerIdentify(myUID, '', true, serverCallMap(dispatch, getState, true)))
     }
   }
 }
 
-export function triggerIdentify (uid: string = '', userAssertion: string = '', allowSelf: boolean = false, incomingCallMap: Object = {},
-  useDelegateUI: boolean = true, allowEmptySelfID: boolean = false, noSkipSelf: boolean = true): TrackerActionCreator {
+const profileReason = 'Profile'
+
+export function triggerIdentify (uid: string = '', userAssertion: string = ''
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: simplify the interface

File: shared/actions/tracker.js [hunk 3]
```diff
@@ -377,6 +386,10 @@ function serverCallMap (dispatch: Dispatch, getState: Function, skipPopups: bool
     start: ({username, sessionID, reason}) => {
       sessionIDToUsername[sessionID] = username
 
+      if (reason && (reason.reason === profileReason)) {
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: we inject a custom reason if we're popping this up so we don't get trackers by accident

File: shared/actions/tracker.js [hunk 4]
```diff
@@ -523,3 +536,87 @@ function updateProof (remoteProof: RemoteProof, linkCheckResult: LinkCheckResult
     payload: {remoteProof, linkCheckResult, username},
   }
 }
+
+function summaryToTrackingInfo (summaries: Array<UserSummary>): Array<TrackingInfo> {
+  return summaries.map(s => ({
+    username: s.username,
+    fullname: s.fullName,
+    following: false, // TODO need to merge in this from a different call
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: will do this next, not in this pr

File: shared/common-adapters/tab-bar.desktop.js [hunk 5]
```diff
@@ -135,7 +135,7 @@ class TabBar extends Component {
 
   render () {
     const tabBarButtons = (
-      <Box style={{...globalStyles.flexBoxRow, ...this.props.styleTabBar}}>
+      <Box style={{...globalStyles.flexBoxRow, flexShrink: 0, ...this.props.styleTabBar}}>
```
Comment (chrisnojima) [inline shared/common-adapters/tab-bar.desktop.js]: don't let this collapse ever

File: shared/nav.desktop.js [hunk 6]
```diff
@@ -188,6 +189,7 @@ class Nav extends Component<void, Props, State> {
     return (
       <MetaNavigator
         tab={tab}
+        globalRoutes={globalRoutes}
```
Comment (chrisnojima) [inline shared/nav.desktop.js]: allow global routes, you can push a profile from anywhere to anywhere

File: shared/profile/index.js [hunk 7]
```diff
@@ -4,25 +4,32 @@ import {connect} from 'react-redux'
...
       subRoutes: {},
     }
   }
 
   componentDidMount () {
-    this.props.refresh()
+    this.props.refresh(this.props.username)
+  }
+
+  shouldComponentUpdate (nextProps, nextState) {
```
Comment (chrisnojima) [inline shared/profile/index.js]: with the nav we have on desktop we don't actually keep the true stack of components , so when you push a profile onto another profile it actually just replaces the data. this makes this all slightly more tricky. we have to keep certain props in the props vs just figuring this out in connect

File: shared/profile/index.js [hunk 8]
```diff
@@ -31,27 +38,36 @@ class Profile extends Component<void, Props, void> {
         showComingSoon={!flags.tabProfileEnabled}
         {...this.props}
         proofs={this.props.proofs || []}
+        onBack={!this.props.profileRoot ? this.props.onBack : undefined}
```
Comment (chrisnojima) [inline shared/profile/index.js]: disallow back on the profile page

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready to look at

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 9252cf8cd47d...f87e2b2f37ee introduced some visual changes:
 * Changed: **Profile-Unfollowed** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Profile-Unfollowed-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Profile-Unfollowed-f87e2b2f37ee.png)  
   Profile-Unfollowed comparison
 * Changed: **Profile-Followed** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Profile-Followed-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Profile-Followed-f87e2b2f37ee.png)  
   Profile-Followed comparison
 * Changed: **Profile-Changed** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Profile-Changed-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Profile-Changed-f87e2b2f37ee.png)  
   Profile-Changed comparison
 * Changed: **Menubar-Public_Truncated** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Menubar-Public_Truncated-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Menubar-Public_Truncated-f87e2b2f37ee.png)  
   Menubar-Public_Truncated comparison
 * Changed: **Menubar-Private_Truncated** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Menubar-Private_Truncated-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Menubar-Private_Truncated-f87e2b2f37ee.png)  
   Menubar-Private_Truncated comparison
 * Changed: **Menubar-Private_Normal** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Menubar-Private_Normal-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Menubar-Private_Normal-f87e2b2f37ee.png)  
   Menubar-Private_Normal comparison
 * Changed: **Menubar-Private_Menu_Showing** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Menubar-Private_Menu_Showing-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466632266779-9252cf8cd47d-f87e2b2f37ee/Menubar-Private_Menu_Showing-f87e2b2f37ee.png)  
   Menubar-Private_Menu_Showing comparison

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 9252cf8cd47d...8d29d34b4ed6 introduced some visual changes:
 * Changed: **Profile-Unfollowed** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466690097395-9252cf8cd47d-8d29d34b4ed6/Profile-Unfollowed-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466690097395-9252cf8cd47d-8d29d34b4ed6/Profile-Unfollowed-8d29d34b4ed6.png)  
   Profile-Unfollowed comparison
 * Changed: **Profile-Followed** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466690097395-9252cf8cd47d-8d29d34b4ed6/Profile-Followed-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466690097395-9252cf8cd47d-8d29d34b4ed6/Profile-Followed-8d29d34b4ed6.png)  
   Profile-Followed comparison
 * Changed: **Profile-Changed** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466690097395-9252cf8cd47d-8d29d34b4ed6/Profile-Changed-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466690097395-9252cf8cd47d-8d29d34b4ed6/Profile-Changed-8d29d34b4ed6.png)  
   Profile-Changed comparison

**Comment by chrisnojima** [comment]
- [ ] have to rebase when https://github.com/keybase/client/pull/3156 is in

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 9252cf8cd47d...933819668b19 introduced some visual changes:
 * Changed: **Profile-Unfollowed** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466693858570-9252cf8cd47d-933819668b19/Profile-Unfollowed-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466693858570-9252cf8cd47d-933819668b19/Profile-Unfollowed-933819668b19.png)  
   Profile-Unfollowed comparison
 * Changed: **Profile-Followed** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466693858570-9252cf8cd47d-933819668b19/Profile-Followed-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466693858570-9252cf8cd47d-933819668b19/Profile-Followed-933819668b19.png)  
   Profile-Followed comparison
 * Changed: **Profile-Changed** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466693858570-9252cf8cd47d-933819668b19/Profile-Changed-9252cf8cd47d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466693858570-9252cf8cd47d-933819668b19/Profile-Changed-933819668b19.png)  
   Profile-Changed comparison

**Pull request merged**